src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -236,7 +236,7 @@ private void checkEnumDef(DetailAST ast) {
         if (isInterfaceOrAnnotationMember(ast)) {
             processInterfaceOrAnnotation(ast);
         }
-        else if (ast.getParent() != null) {
+        else {
             checkForRedundantModifier(ast, TokenTypes.LITERAL_STATIC);
```

Comment:
An enum can't be a top class and have `static` modifier. It won't compile.
This check may prevent some excess computations from happening since we now it can't be possible. Should we still remove this check? The only way to kill it would be to use powermock to count calls to `checkForRedundantModifier`.

Reply:
Let's remove